flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/debezium/DebeziumUtils.java

```diff
@@ -324,11 +324,22 @@ private static long getBinlogTimestamp(BinaryLogClient client, String binlogFile
             client.setBinlogFilename(binlogFile);
             client.setBinlogPosition(0);
 
-            LOG.info("begin parse binlog: {}", binlogFile);
+            LOG.info("Begin parse binlog: {}", binlogFile);
             client.connect();
         } finally {
             client.unregisterEventListener(eventListener);
         }
+        if (binlogTimestamps.isEmpty()) {
+            try {
+                if (client.isConnected()) {
+                    client.disconnect();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            LOG.warn("Failed to register eventListener and try to register it again");
+            return getBinlogTimestamp(client, binlogFile);
```

Comment:
What about using loop instead of recursion?

Reply:
> What about using loop instead of recursion?

all the same .

Reply:
Just worrying about overflow of calling stack. But since connection failure not very likely to happen, I'll not insist about this.

Reply:
> Just worrying about overflow of calling stack. But since connection failure very unlikely to happen, I'll not insist on this.

Yes, you're right. I will adopt your opinion